entry/src/entry.rs

```diff
@@ -111,15 +111,24 @@ pub struct Api<'a> {
 /// field points to Transactions that took place shortly before `hash` was generated.
 ///
 /// If you multiply `num_hashes` by the amount of time it takes to generate a new hash, you
-/// get a duration estimate since the last Entry. Since processing power increases
+/// get a duration estimate since the last `Entry`. Since processing power increases
 /// over time, one should expect the duration `num_hashes` represents to decrease proportionally.
 /// An upper bound on Duration can be estimated by assuming each hash was generated by the
 /// world's fastest processor at the time the entry was recorded. Or said another way, it
 /// is physically not possible for a shorter duration to have occurred if one assumes the
 /// hash was computed by the world's fastest processor at that time. The hash chain is both
 /// a Verifiable Delay Function (VDF) and a Proof of Work (not to be confused with Proof of
 /// Work consensus!)
-
+///
+/// The solana core protocol currently requires an `Entry` to contain `transactions` that are
+/// executable in parallel (implemented in
+/// `solana_core::banking_stage::BankingStage::process_and_record_transactions()`). All transactions
+/// in the `transactions` field have to follow the read/write locking restrictions with regard to
+/// the accounts they reference. A single account can be either written by a single transaction, or
+/// read by one or more transactions, but not both.
+///
+/// This enforcement is done via a call to `solana_runtime::accounts::Accounts::lock_accounts()`
+/// with the `txs` argument holding all the `transactions` in the `Entry`.
```

Comment:
I think you are looking at this from the TPU standpoint.
For TVUs, `Entry`es is what they receive.  So the transactions are already there.
Are you sure you want me to change like this?

Reply:
Yeah, true. But from the TPU standpoint either, "in the Entry" feels awkward because the Entry doesn't exist at the time `lock_accounts` is called. Oh well. It's probably clear enough to most readers.